lib/filterx/object-primitive.c

```diff
@@ -139,12 +139,46 @@ _double_add(FilterXObject *s, FilterXObject *object)
 }
 
 gboolean
-double_repr(double val, GString *repr)
+double_repr(double dbl, GString *repr)
 {
-  gsize init_len = repr->len;
-  g_string_set_size(repr, init_len + G_ASCII_DTOSTR_BUF_SIZE);
-  g_ascii_dtostr(repr->str + init_len, G_ASCII_DTOSTR_BUF_SIZE, val);
-  g_string_set_size(repr, init_len + strlen(repr->str + init_len));
+  if (isnan(dbl))
+    {
+      g_string_append_len(repr, "NaN", 3);
+    }
+  else if (isinf(dbl))
+    {
+      if (dbl < 0)
+        g_string_append_c(repr, '-');
+      g_string_append_len(repr, "Infinity", 8);
```

Comment:
Both JSON RFCs says:
```
Numeric values that cannot be represented in the grammar below (such as Infinity and NaN) are not permitted.
```

A few languages/libraries serialize these values incorrectly.
If we want to represent these, I think we need to use strings instead.

Reply:
snprintf() also produces "inf" in case the result is infinite.

Reply:
We discussed that there is no good way to resolve this issue. A few programming languages and libraries decided to violate the RFC as well because in very generic cases, there is no good choice in such cases: encode these values as strings? skip the keys that hold such values?